src/shadowbox/integration_test/test.sh

```diff
@@ -81,6 +81,29 @@ function cleanup() {
   return $status
 }
 
+# Runs a test command and checks for expected result code and retries on failure.
+# Use for tests that are idempotent and flaky.
+function test_with_retries() {
+  local test_command="${1}"             # Test command to run
+  local expected_result_code="${2}"     # Result code to expected for test command
+  local failure_message="${3}"          # Message to display on failure
+  local max_attempts="${4:-5}"          # Maximum number of attempts
+
+  local RETRY_DELAY_SECONDS=5
+  local attempt_count=1
+
+  until (${test_command} > /dev/null || (($? == ${expected_result_code}))); do
```

Comment:
Use a for loop instead: https://linuxize.com/post/bash-for-loop/#the-c-style-bash-for-loop

Reply:
Done